tests/manage/pv_services/pv_encryption/test_restore_snapshot_for_encrypted_or_unencrypted_pvc_to_unencrypted_or_encrypted_sc.py

```diff
@@ -0,0 +1,502 @@
+            md5sum_after_io = cal_md5sum(
+                pod_obj=pod_obj,
+                file_name=pod_obj.get_storage_path(storage_type="block"),
+                block=True,
+            )
+            assert (
+                pod_obj.md5sum_before_io != md5sum_after_io
+            ), f"md5sum has not changed after IO on pod {pod_obj.name}"
+
+            # Create snapshot
+            log.info(f"Creating snapshot of PVC {pod_obj.pvc.name}")
+            snap_obj = snapshot_factory(pod_obj.pvc, wait=False)
+            snap_obj.md5sum = md5sum_after_io
+            snap_objs.append(snap_obj)
+            log.info("Snapshots created")
+
+        # Verify snapshots are ready
+        log.info("Verify snapshots are ready")
+        for snap_obj in snap_objs:
+            snap_obj.ocp.wait_for_resource(
+                condition="true",
+                resource_name=snap_obj.name,
+                column=constants.STATUS_READYTOUSE,
+                timeout=180,
+            )
+            snapshot_content = get_snapshot_content_obj(snap_obj=snap_obj)
+            snap_handle = snapshot_content.get().get("status").get("snapshotHandle")
+            snap_handles.append(snap_handle)
+
+        # Restoring snapshots to create new PVCs
+        log.info("Creating new PVCs from snapshots")
+        restore_pvcs = []
+        for snap_obj in snap_objs:
+            # for storageclass in sc_objs:
+            log.info(f"Creating a PVC from snapshot {snap_obj.name}")
+            restore_pvc_obj = snapshot_restore_factory(
+                snapshot_obj=snap_obj,
+                storageclass=self.sc_obj_with_encryption.name,
+                size=f"{self.pvc_size}Gi",
+                volume_mode=snap_obj.parent_volume_mode,
+                access_mode=snap_obj.parent_access_mode,
+                status="",
+            )
+            restore_pvcs.append(restore_pvc_obj)
+
+            log.info(
+                f"Created PVC {restore_pvc_obj.name} from snapshot {snap_obj.name}."
+                f"Used the storage class {self.sc_obj_with_encryption}"
+            )
+            restore_pvc_obj.md5sum = snap_obj.md5sum
+            failure_str = "cannot create encrypted volume from unencrypted volume"
+            # Confirm that the restored PVCs are in Pending status
+            for restore_pvc_obj in restore_pvcs:
+                helpers.wait_for_resource_state(
+                    resource=restore_pvc_obj,
+                    state=constants.STATUS_PENDING,
+                    timeout=200,
+                )
+                restore_pvc_obj.reload()
+                if failure_str in restore_pvc_obj.describe():
+                    log.info(
+                        f"cannot create encrypted volume from unencrypted volume {snap_obj.name}"
+                    )
+                else:
+                    log.warning(
+                        f"Able to create encrypted volume from unencrypted volume {snap_obj.name}"
```

Comment:
This should be logged as an error instead of a warning

Reply:
Updated to error